fn/api.go

```diff
@@ -21,11 +21,20 @@ func host() string {
 		log.Fatalln("Couldn't parse API URL:", err)
 	}
 
-	return u.Host
+	return u
+}
+
+func host() string {
+	return apiURL().Host
+}
+
+func scheme() string {
+	return apiURL().Scheme
 }
 
 func apiClient() *fnclient.Functions {
-	transport := httptransport.New(host(), "/v1", []string{"http"})
+	transport := httptransport.New(host(), "/v1", []string{scheme()})
```

Comment:
It seems a little redundant and racy to get and parse the url twice in a row. What about calling `apiURL` once from here and accessing the `Host` and `Scheme` fields directly?